styles/style_ptr.h

```diff
@@ -3,6 +3,17 @@
 
 #include "blade_style.h"
 
+template<class BASE, class L1> class Compose;  // forward-declare Layers node
+
+namespace style_base_check_detail {
+  template<class T> struct TopBase { using type = T; };
+  template<class B, class L> struct TopBase<Compose<B,L>> : TopBase<B> {};
+
+  template<typename T> struct IsOpaqueColor : std::false_type {};
```

Comment:
This is the same code as in layers_details, so it should definitely go in common/color.h